src/js/components/Pagination/doc.js

```diff
@@ -0,0 +1,98 @@
+import { describe, PropTypes } from 'react-desc';
+
+import { genericProps } from '../../utils/prop-types';
+import { getAvailableAtBadge } from '../../utils/mixins';
+import { themeDocUtils } from '../../utils/themeDocUtils';
+
+export const doc = Pagination => {
+  const DocumentedPagination = describe(Pagination)
+    .availableAt(getAvailableAtBadge('Pagination'))
+    .description(
+      `A control that enables selection of a single page from a 
+      range of pages.`,
+    )
+    .usage(
+      `import { Pagination } from 'grommet';
+      <Pagination />`,
+    )
+    .intrinsicElement('nav');
+
+  DocumentedPagination.propTypes = {
+    ...genericProps,
+    numEdgePages: PropTypes.number
+      .description(
+        `The number of page buttons visible at the start and end of page 
+        range.`,
+      )
+      .defaultValue(1),
+    numItems: PropTypes.number
+      .description('The number of items to paginate.')
+      .defaultValue(undefined),
+    numMiddlePages: PropTypes.number
+      .description(
+        `The number of page buttons visible in the middle of the controls.`,
+      )
+      .defaultValue(3),
+    onChange: PropTypes.func
+      .description(
+        `Function that will be called when the user clicks a page or 
+        arrow button. The single argument is an event containing the latest 
+        page via \`event.page\`, and the startIndex and endIndex for data on 
+        this page via \`event.startIndex\` and \`event.endIndex\`, 
+        respectively.`,
+      )
+      .defaultValue(undefined),
+    page: PropTypes.number
+      .description(
+        `The default page. If used with onChange, it can be used to control the 
+        active page via state.`,
+      )
+      .defaultValue(undefined),
+    step: PropTypes.number
+      .description('The number of items per page.')
+      .defaultValue(10),
+  };
+
+  return DocumentedPagination;
+};
+
+export const themeDoc = {
+  pagination: {
+    description: `The possible sizes of the Pagination in terms of its 
+      max-width, font-size and line-height.`,
```

Comment:
according to the description, any reason not to use `pagination.size` instead of `pagination`? to be more consistent with theme props.